### PR TITLE
daemon: allow polkit authentication for `/v2/snaps/{snap}/conf`

### PR DESCRIPTION
**Pull request by olivercalder**
For prompting, non-root clients need to be able to query and set the `system experimental.apparmor-prompting` option. Previously, there was no good way to do so without a workaround in the client to make a query as root, but using polkit is much preferred. This PR enables polkit authentication on the `/v2/snaps/{snap}/conf` endpoint.

Relevant discussion: https://forum.snapcraft.io/t/get-on-authenticated-endpoints-fail-despite-x-allow-interaction/39067

### CODE REVIEW

**Review by alexmurray — APPROVED** [review]
Quoting from https://github.com/snapcore/snapd/pull/13367#pullrequestreview-1897445367 

> The polkit policy for `io.snapcraft.snapd.manage` currently requires `auth_admin` so this is effectively equivalent to root which is what the bare `authenticatedAccess{}` checks when no polkit action is specified. As such there is no material change to the security impact from this change. +1 from me.

**Review by pedronis — APPROVED** [review]
thanks

**Comment by pedronis** [comment]
@olivercalder we should consider though doing something like we do for interfaces though, as otherwise we need to extend the description even futher which might be a bit too much

**Review by zyga — COMMENTED** [review]
I think having an explicit new polkit action for managing system wide snap configuration is indeed clearer, as we can have a nicer message in the prompt and, well, it's just more scoped.

Shall we add something like this to `daemon/api.go`:
``` go
const polkitActionManageConfiguration = "io.snapcraft.snapd.manage-configuration"
```
and something like this to `data/polkit/io.snapcraft.snapd.policy`:

``` xml
  <action id="io.snapcraft.snapd.manage-configuration">
    <description gettext-domain="snappy">Access or modify snap configuration</description>
    <message gettext-domain="snappy">Authentication is required to access or modify snap configuration</message>
    <defaults>
      <allow_any>auth_admin</allow_any>
      <allow_inactive>auth_admin</allow_inactive>
      <allow_active>auth_admin_keep</allow_active>
    </defaults>
  </action>
```

File: daemon/api_snap_conf.go [hunk 1]
```diff
@@ -38,8 +38,8 @@ var (
 		Path:        "/v2/snaps/{name}/conf",
 		GET:         getSnapConf,
 		PUT:         setSnapConf,
-		ReadAccess:  authenticatedAccess{},
-		WriteAccess: authenticatedAccess{},
+		ReadAccess:  authenticatedAccess{Polkit: polkitActionManage},
+		WriteAccess: authenticatedAccess{Polkit: polkitActionManage},
```
Comment (jhenstridge) [inline daemon/api_snap_conf.go:42]: I'd suggest using a new action ID with a description that matches the type of access that is being granted. The new action should be defined in `data/polkit/io.snapcraft.snapd.policy`.

Also, what sort of user interface do you expect to see for this? If it was a toggle switch displaying whether AppArmor prompting was enabled that the user could turn on and off, authenticated read access means we'd show a polkit prompt just by displaying the UI before the user has decided to change anything. That seems like it'd be a bit confusing.

Comment (pedronis) [reply daemon/api_snap_conf.go:42]: For the read side we plan to separately add information about which experimental features are supported, respectively enabled to the /v2/system-info endpoint which is not restricted. cc @olivercalder

**Comment by olivercalder** [comment]
Apologies for force-pushing, I wanted to get the new polkit action defined first.

**Review by pedronis — APPROVED** [review]
thanks

**Comment by pedronis** [comment]
@jhenstridge @alexmurray please give this a look again

**Review by alexmurray — APPROVED** [review]
LGTM!

**Comment by olivercalder** [comment]
The ubuntu-core-18 test fails consistently with a corrupt system error early on, I don't see any reason to believe it's related to the changes here. Will likely need a force push, unfortunately.

**Review by jhenstridge — APPROVED** [review]
Looks good!

**Pull request merged**